### DISCUSSION TITLE
What Network Library?

### DISCUSSION THREAD

**Discussion by jasonswearingen** [stride3d/stride#1115] [Ideas]
For basic background on networking in games and the various strategies, see https://www.gafferongames.com/post/what_every_programmer_needs_to_know_about_game_networking/  and the related articles

Some options:

1. Networking via SteamWorks library:
   - https://partner.steamgames.com/doc/features/multiplayer/networking
   - a good C# binding: https://github.com/Facepunch/Facepunch.Steamworks
2. Networking via Epic Online Services:
   - https://dev.epicgames.com/en-US/services
   - official C# SDK:  https://dev.epicgames.com/docs/services/en-US/CSharp/GettingStarted/index.html
3. Litenetlib, a spiritual successor to the defunct Lidgren network libraries: https://github.com/RevenantX/LiteNetLib
4. A library supporting Nat Traversal: https://github.com/Toemsel/Network
5. A C# Port of the ENet Library:
   - https://github.com/nxrighthere/ENet-CSharp
   - https://github.com/lsalzman/enet  (original lib)
   - http://enet.bespin.org/ (docs for ENet)
6. Simple implementation directly tied to Stride: https://github.com/manio143/Stride.Networking.Simple 

Other networking related readings/research:
- NAT
  - C# NAT library: https://github.com/lontivero/Open.NAT
- Cheaters (other good networking stuff on site/book too)
  - http://ithare.com/chapter-iii-on-cheating-p2p-and-non-authoritative-servers-from-dd-of-mmog-book/
- other libraries not yet researched
   - https://github.com/Toemsel/Network
   - https://github.com/RevenantX/LiteNetLib

**Comment by danilobreda** [comment]
https://github.com/RevenantX/LiteNetLib seems more updated and strong.

**Reply by MsEpsilon** [reply]
I personally use it, together with LiteEntitySystem. 👍
